content/rs/developing/crdbs/developing-streams-crdb.md

```diff
@@ -1,278 +1,320 @@
-"120-0"
-127.0.0.1:6379> XADD x 130 f v
-"130-0"
-127.0.0.1:6379> XADD x 140 f v
-"140-0"
-127.0.0.1:6379> XADD x 150 f v
-"150-0"
-127.0.0.1:6379> XREAD COUNT 2 STREAMS x 0
-1) 1) "x"
-   2) 1) 1) "110-0"
-         2) 1) "f"
-            2) "v"
-      2) 1) "120-0"
-         2) 1) "f"
-            2) "v"
-127.0.0.1:6379> XREAD COUNT 2 STREAMS x 120
-1) 1) "x"
-   2) 1) 1) "130-0"
-         2) 1) "f"
-            2) "v"
-      2) 1) "140-0"
-         2) 1) "f"
-            2) "v"
-127.0.0.1:6379> XREAD COUNT 2 STREAMS x 140
-1) 1) "x"
-   2) 1) 1) "150-0"
-         2) 1) "f"
-            2) "v"
+Set _strict_ mode:
+```sh
+rladmin> tune db crdb crdt_xadd_id_uniqueness_mode strict
 ```
 
-Using XREAD is not recommended as an iterator in Active-Active databases because entries can be skipped:
+Set _semi-strict_ mode:
+```sh
+rladmin> tune db crdb crdt_xadd_id_uniqueness_mode semi-strict
+```
+
+Set _liberal_ mode:
+```sh
+rladmin> tune db crdb crdt_xadd_id_uniqueness_mode liberal
+```
+
+### Iterating a stream with XREAD
+
+In open source Redis and in non-Active-Active databases, you can use XREAD to iterate over the entries in a Redis Stream. However, with an Active-Active database, XREAD may skip entries. This can happen when multiple regions write to the same stream.
 
-| Time | Location 1                                         | Location 2                                         |
+In the example below, XREAD skips entry `115-0`.
+
+| Time | Region 1                                         | Region 2                                         |
 | ---- | -------------------------------------------------- | -------------------------------------------------- |
-| t1   | XADD x 110 f1 v1                                   | XADD x 115 f1 v1                                   |
-| t2   | XADD x 120 f1 v1                                   |                                                    |
-| t3   | XADD x 130 f1 v1                                   |                                                    |
-| t4   | XREAD COUNT 2 STREAMS x 0 --> [110-0, 120-0]       |                                                    |
-| t5   | --- sync ---                                       | --- sync ---                                       |
-| t6   | XREAD COUNT 2 STREAMS x 120 --> [130-0]            |                                                    |
-| t7   | XREAD STREAMS x 0 --> [110-0, 115-0, 120-0, 130-0] | XREAD STREAMS x 0 --> [110-0, 115-0, 120-0, 130-0] |
+| _t1_   | `XADD x 110 f1 v1`                                   | `XADD x 115 f1 v1`                                   |
+| _t2_   | `XADD x 120 f1 v1`                                   |                                                    |
+| _t3_   | `XADD x 130 f1 v1`                                   |                                                    |
+| _t4_   | `XREAD COUNT 2 STREAMS x 0` <br/>**→ [110-2700, 120-2700]**       |                                                    |
```

Comment:
Any reason for using so large seq. IDs? I find it confusing

Reply:
+ my comment on the SEQ part (need to be %32)